hibernate-core/src/test/java/org/hibernate/orm/test/query/hhh15645/HHH15645Test.java

```diff
@@ -0,0 +1,77 @@
+package org.hibernate.orm.test.query.hhh15645;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.orm.junit.EntityManagerFactoryScope;
+import org.hibernate.testing.orm.junit.Jpa;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Join;
+import jakarta.persistence.criteria.JoinType;
+import jakarta.persistence.criteria.ListJoin;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@TestForIssue(jiraKey = "HHH-14916")
+@Jpa(
+		annotatedClasses = { HHH15645Test.TestParent.class, HHH15645Test.TestChild.class}
+)
+public class HHH15645Test {
+
+	@Test
+	public void test(EntityManagerFactoryScope scope) {
+		scope.inTransaction( entityManager -> {
+			final TestParent p = new TestParent();
+			p.setId(1);
+			entityManager.persist(p);
+
+			final CriteriaBuilder cb = entityManager.getCriteriaBuilder();
+			final CriteriaQuery<TestChild> c = cb.createQuery(TestChild.class);
+			final Root<TestChild> r = c.from(TestChild.class);
+			final Join<TestChild, TestParent> g = r.join( "parent" );
```

Comment:
Note: If I change the join to use `JoinType.LEFT`, the test passes for some reason.